enhancements/ingress/custom-route-configuration.md

```diff
@@ -72,82 +76,97 @@ slightly different documentation.
+    // ensures that no two components will conflict and the same component can be installed multiple times.
     Namespace string
     // name is the *logical* name of the route to customize.  It does not have to be the actual name of a route resource.
     // Keep in mind that this is your API for users to customize.  You could later rename the route, but you cannot rename
     // this name.
     Name string
-    // defaultHost is the normal name of this route.
-    DefaultHost string
+    // defaultHostname is the normal host name of this route.  It is provided in case cluster-admins find it more recognizeable
+    // and having it here makes it possible to answer, "if I remove my configuration, what will the name be".
+    DefaultHostname string
     // ConsumingUsers is a slice of users that need to have read permission on the secrets in order to use them.
     // This will usually be an operator service account.
     ConsumingUsers []string
-    // currentHost is the current name used to by the route.  Routes can have more than one exposed name, even though we
+    // currentHostnames is the current name used to by the route.  Routes can have more than one exposed name, even though we
     // only allow one route.spec.host.
-    CurrentHosts []string
+    CurrentHostnames []string
 
-    // conditions are available, degraded, progressing.  This allows consistent reporting back and feedback that is well
+    // conditions are degraded and progressing.  This allows consistent reporting back and feedback that is well
     // structured.  These particular conditions have worked very well in ClusterOperators.
+    // Degraded == true means that something has gone wrong trying to handle the ComponentRoute.  The CurrentHostnames
+    // may or may not be operating successfully.
+    // Progressing == true means that the component is taking some action related to the ComponentRoute
     Conditions []ConfigCondition
 
     // relatedObjects allows listing resources which are useful when debugging or inspecting how this is applied.
-    // They may be aggregated into an overal status RelatedObjects to be automatically by oc adm inspect
+    // They may be aggregated into an overall status RelatedObjects to be automatically shown by oc adm inspect
     RelatedObjects []ObjectReference
     
     
 
     // This API does not include a mechanism to distribute trust, since the ability to write this resource would then 
     // allow interception.  Instead, if we need such a mechanism, we can talk about creating a way to allow narrowly scoped
-    // updates to a configmap containing ca-bundle.crt for each CustomizeableRoute.
+    // updates to a configmap containing ca-bundle.crt for each ComponentRoute.
     // CurrentCABundle []byte
 }
 ```
 
+Validation rules to be specified in the openshift/api PR, but basic restrictions on strings as you'd expect.
```

Comment:
> These all are not necessarily API validation of the object, but something that should still exist in the operator before it pushes whatever suspicious configuration it is supplied. Some of these checks might make more sense even after the configuration is pushed on.

You cannot do these checks at API validation time, since references can change and the system is supposed to be eventually consistent with creation in any order.  I'm willing to recommend, "don't apply stuff that's broken", but it must not be done at API validation time because other changes (like secret updates) can take the content in an unworkable direction.

Reply:
Sure, my point is that we should add these recommendations as to how an operator should treat this new configuration before and after applying it, and how to report problems it finds.